### ISSUE TITLE
S1940: FP with null-conditional operator

### ISSUE THREAD

**Issue by k0balt** [SonarSource/sonar-dotnet#2427]
### Repro steps

```cs
public static bool IsNullOrEmpty<T>(this IList<T> collection)
    => !(collection?.Count > 0);
```

### Expected behavior

Rule S1940 should not triggered

### Actual behavior

On example above sonar suggest replace `!(collection?.Count > 0)` with `collection?.Count <= 0`, but if `collection == null` conditions are not equal. 
In same time, in this code:
```cs
public static bool IsNotPositive(this int? num)
    => !(num > 0);
```
Sonar doesn't suggest replacement (and that's right behavior).

### Related information

* SonarC# Version 4.9.0.4316
* Visual Studio Version 15.9.12

**Comment by andrei-epure-sonarsource** [comment]
thanks @k0balt for the detailed reproducer
